### PR TITLE
fix: add missing file properties on Windows, closes #6676

### LINKED ISSUE #6676: [bug] Copyright not populating properly
### Describe the bug

When building the Tauri exe the copyright field does not populate from bundle to the windows .exe

image
image

### Reproduction

Build an application with the bundle flag for copyright
check for copyright after build

### Expected behavior

the copyright field to reflect what was in the bundle json

### Platform and versions

```text
output of cargo tauri build
```

### Stack trace

_No response_

### Additional context

_No response_

### LINKED ISSUE #6870: [bug] `publisher` field not working with NSIS target
### Describe the bug

When bundling for Windows, the msi package uses the `publisher` field of the config file but the NSIS package does not.

`tauri.conf.json`
```json
{
  ...
  "tauri": {
    ...
    "bundle": {
      ...
      "publisher": "dubis.dev",
      "identifier": "dev.dubis.runmath",
      "targets": [
        "msi",
        "nsis"
      ]
    },
    ...
}
```

Results:

⬇️ MSI
image
🔼 NSIS

### Reproduction

_No response_

### Expected behavior

Both fields should be the same

### Platform and versions

```text
[✔] Environment
    - OS: Windows 10.0.22621 X64
    ✔ WebView2: 112.0.1722.68
    ✔ MSVC: Visual Studio Build Tools 2022
    ✔ rustc: 1.66.0 (69f9c33d7 2022-12-12)
    ✔ Cargo: 1.66.0 (d65d197ad 2022-11-15)
    ✔ rustup: 1.25.1 (bb60b1e89 2022-07-12)
    ✔ Rust toolchain: stable-x86_64-pc-windows-msvc (default)
    - node: 18.14.2
    - pnpm: 8.2.0
    - yarn: 1.22.19
    - npm: 9.6.6

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.1
    - tao [RUST]: git+https://github.com/tauri-apps/tao?branch=v0.16#ed6aab6b1226e56683798732ad01bf3f4658097e (0.16.0)
    - @tauri-apps/api [NPM]: 1.3.0
    - node: 18.14.2
    - pnpm: 8.2.0
    - yarn: 1.22.19
    - npm: 9.6.6

[-] Packages
    - tauri [RUST]: 1.3.0
    - tauri-build [RUST]: 1.3.0
    - wry [RUST]: 0.24.1
    - tao [RUST]: git+https://github.com/tauri-apps/tao?branch=v0.16#ed6aab6b1226e56683798732ad01bf3f4658097e (0.16.0)
    - @tauri-apps/api [NPM]: 1.3.0
    - @tauri-apps/cli [NPM]: 1.3.0

[-] App
```

### Stack trace

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [x] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/bundler/src/bundle/windows/nsis.rs [hunk 1]
```diff
@@ -118,6 +118,25 @@ fn get_and_extract_nsis(nsis_toolset_path: &Path, _tauri_tools_path: &Path) -> c
...
+  let version = semver::Version::parse(version_str).context("invalid app version")?;
+  if !version.build.is_empty() {
+    let build = version.build.parse::<u64>();
+    if build.is_ok() {
+      return Ok(format!(
+        "{}.{}.{}.{}",
+        version.major, version.minor, version.patch, version.build
+      ));
+    } else {
+      anyhow::bail!("optional build metadata in app version must be numeric-only and cannot be greater than 65535 for msi target");
```
Comment (amrbashir) [inline tooling/bundler/src/bundle/windows/nsis.rs]:
```suggestion
      anyhow::bail!("optional build metadata in app version must be numeric-only");
```

**Review by wusyong — DISMISSED** [review]

**Comment by FabianLars** [comment]
i don't think the issue about the publisher field will be resolved by this. There is no new reference to it. The only thing i found that looked related was adding something like `VIAddVersionKey "PCompanyName" "${MANUFACTURER}"` but i'm not sure if that's really the same thing.

**Comment by amrbashir** [comment]
@FabianLars  `VIAddVersionKey` is for the installer not the resulting app, the publisher for the app is defined later in the script using registry.

This is the fix for the publisher field https://github.com/tauri-apps/tauri/pull/6693/files#diff-ce48fea6a58a4b6b159ffed70c017f279effea3f065cd064f38fba42e93af25fR183-R185
we have `tauri > bundle > windows > publisher` filed but it wasn't respected on Windows and instead used the middle part of the identifier, now it will use the designated field and fallback to the middle part of the identifier.

**Comment by FabianLars** [comment]
ah mb, i thought the linked issue was about the installer's file name. Now on a second look i see that the screenshot even includes memory usage so that doesn't make sense, sorryyy :v:

File: .changes/tauri-build-shortdesc-copyright.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'tauri-build': 'patch'
+---
+
+On Windows, set `Copyright` and `FileDescription` file properties on the executable from `tauri.bundle.copyright` and `tauri.bundle.shortDescription`,
```
Comment (lucasfernog) [inline .changes/tauri-build-shortdesc-copyright.md]:
```suggestion
On Windows, set `LegalCopyright` and `FileDescription` file properties on the executable from `tauri.bundle.copyright` and `tauri.bundle.shortDescription`,
```

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**